### PR TITLE
FEATURE: Set meta:"*" to get a map of all keys and values

### PR DESCRIPTION
**Pull request by austintaylor**

### CODE REVIEW

**Comment by cypriss** [comment]
@austintaylor What is the use-case here?

Curious if you considered the alternative of something like:

```
type withMetaStar struct {
    AField    String
    CustomFields StringMap
 }
```

Obviously the latter isn't as flexible, so I'm curious if your use-case requires some of the flexibility that you'd lose.

File: meta_test.go [hunk 1]
```diff
@@ -120,4 +120,22 @@ func TestValuers(t *testing.T) {
...
+	AllFields map[string]interface{} `meta:"*"`
+}
+
+var withMetaStarDecoder = NewDecoder(withMetaStar{})
+
+func TestMetaStar(t *testing.T) {
+	var inputs withMetaStar
+	e := withMetaStarDecoder.Decode(&inputs, url.Values{"cf_other_field": {"Another field"}}, []byte(`{"a_field": "A field", "cf_numeric_field": 12}`))
+	assertEqual(t, e, ErrorHash(nil))
+	assertEqual(t, inputs.AField.Val, "A field")
```
Comment (taylorchu) [inline meta_test.go:134]: assertEqual argument order matters here.

**Comment by taylorchu** [comment]
I also don't know the context why we need this. @austintaylor

**Comment by austintaylor** [comment]
We did consider having a separate container for custom fields, but we would prefer to have them on the same level as the standard fields. (We do require custom fields to have a prefix to prevent name collisions.)

```
{
  "a_field": "value",
  "cf_my_custom_field": "custom value"
}
```

In general it seems useful to me to have this kind of escape hatch. This way I can continue to use meta to process simple params, and I can handle more complex cases myself without meta getting in the way.

**Pull request merged**